src/modules/nathelper/nathelper.c

```diff
@@ -2445,6 +2452,175 @@ static int sel_rewrite_contact(str *res, select_t *s, struct sip_msg *msg)
 
 	return 0;
 }
+/**
+* wrapper of set_alias_to_avp_f
+*/
+static int ki_set_alias_to_avp(struct sip_msg *msg, char *uri_avp){
+	str dest_avp={0,0};
+
+	if(!uri_avp)
+	       return -1;
+
+	dest_avp.s=uri_avp;
+	dest_avp.len=strlen(dest_avp.s);
+
+	return set_alias_to_avp_f(msg,&dest_avp);
+
+}
+/**
+* function reads from @param msg then write to given avp @param uri_avp
+* as sip uri
+* @result 1 succes , -1 fail
+* @param msg sip message
+* @param uri_avp given avp name
+*/
+static int set_alias_to_avp_f(struct sip_msg *msg, str *uri_avp){
+	str contact;
+	str alias_uri={0,0};
+
+
+	if(parse_headers(msg,HDR_CONTACT_F,0) < 0 ) {
+	       LM_ERR("Finding Contact Header is failed\n");
+	       return -1;
+	}
+
+	if(!msg->contact)
+	       return -1;
+
+	if(parse_contact(msg->contact)<0 || !msg->contact->parsed ||
+	((contact_body_t *)msg->contact->parsed)->contacts==NULL ||
+	((contact_body_t *)msg->contact->parsed)->contacts->next!=NULL){
+	       LM_ERR("Parsing Contact Header is failed\n");
+	       return -1;
+	}
+
+
+
+	contact.s = ((contact_body_t *)msg->contact->parsed)->contacts->name.s;
+	contact.len = ((contact_body_t *)msg->contact->parsed)->contacts->len;
+
+	alias_to_uri(&contact,&alias_uri);
+	write_to_avp(msg, &alias_uri, uri_avp);
+
+	return 1;
+
+}
+/**
+* write_to_avp function writes data to given avp
+* @param data  source data
+* @param uri_avp destination avp name
+*/
+void write_to_avp(struct sip_msg *msg, str *data, str *uri_avp){
+	pv_spec_t *pvresult = NULL;
+	pv_value_t valx;
+	pvresult = pv_cache_get(uri_avp);
+
+	if(pvresult == NULL) {
+	       LM_ERR("Failed to malloc destination pseudo-variable \n");
+	       return;
+	}
+
+	if(pvresult->setf==NULL) {
+	       LM_ERR("destination pseudo-variable is not writable: %.*s \n",uri_avp->len, uri_avp->s);
+	       return;
+	}
+	memset(&valx, 0, sizeof(pv_value_t));
+
+	if(!data->s){
+				 LM_ERR("There isnt any data to write avp\n");
+				 return;
+	}
+
+	valx.flags      =       PV_VAL_STR;
+	valx.rs.s       =       data->s;
+	valx.rs.len     =       data->len;
+
+	LM_DBG("result: %.*s\n", valx.rs.len, valx.rs.s);
+	pvresult->setf(msg, &pvresult->pvp, (int)EQ_T, &valx);
+
+}
+/**
+* function alias_to_uri  select alias paramter from @param contact_header
+* then writes to @param alias_uri
+* @param contact_header  Source contact header
+* @param alias_uri destination string
+*/
+void alias_to_uri(str *contact_header, str *alias_uri){
```

Comment:
Same as above, consider to return an error in case this function fails to properly evaluate it in the calling function.